tide_media.install

```diff
@@ -150,3 +150,81 @@ function tide_media_update_8002() {
     }
   }
 }
+
+/**
+ * Video transcript field change from text field to WYSIWYG
+ */
+function tide_media_update_8003() {
+  $database = Drupal::database();
+
+  //step1 create two temporary tables for holding data
+  $tmp_table_names = [
+    'tmp_table' => 'media__field_media_transcript',
+    'tmp_revision_table' => 'media_revision__field_media_transcript',
+  ];
+  $spec = [
+    'type' => 'text',
+    'size' => 'big',
+    'not null' => FALSE,
+  ];
+
+  foreach ($tmp_table_names as $tmp => $table_name) {
+    if ($database->schema()->tableExists($table_name)){
+      //creates two temporary tables for holding the data
+      $database->query("CREATE TABLE {$tmp} SELECT * FROM  {$table_name}");
+      $database->truncate($table_name)->execute();
+    }
+  }
+
+  //step2 update field config from string_long to text_long
+  $fields = [
+    'field_media_transcript' => [
+      'table' => 'media__field_media_transcript',
+      'revision_table' => 'media_revision__field_media_transcript',
+      'format_col' => 'field_media_transcript_format',
+    ],
+  ];
+
+  foreach ($fields as $field_name => $f) {
+    $entity_type = 'media';
+    $field_storage = FieldStorageConfig::loadByName($entity_type, $field_name);
+
+    if (is_null($field_storage)) {
+      continue;
+    }
+
+    $new_fields = [];
+    foreach ($field_storage->getBundles() as $bundle => $label) {
+      $field = FieldConfig::loadByName($entity_type, $bundle, $field_name);
+      $new_field = $field->toArray();
+      $new_field['field_type'] = 'text_long';
+      $new_fields[] = $new_field;
+      $field->delete();
```

Comment:
@vincent-gao as the field tables are truncated, the field itself has no data. You should be able to change its field type directly without deleting and re-creating the field.

Reply:
Hi @sonnykt ,  if the UUID didn't change, we don't need to keep fixating on `change it field type`.

Reply:
@vincent-gao I'm hesitate to delete fields unless it's necessary. Deleting a field may cause its dependants being updated/deleted. For example, deleting a field will **disable** all views using that view in Fields, Filters, Relationships, etc. As Tide Media is an opensource project, we can't be sure if there is any dependants of the field in other independent sites.

<img width="649" alt="image" src="https://user-images.githubusercontent.com/167788/58685317-a95d4e00-83be-11e9-9fa5-8fa13a7e1d07.png">

Reply:
Also, if you delete the field, it's pointless to truncate the field tables as they will be deleted anyway.